client_test.go

```diff
@@ -6229,4 +6258,30 @@ func TestGocloak_CreateAuthenticationFlowsAndCreateAuthenticationExecution(t *te
 		}
 	}
 	require.True(t, deleted, "Failed to delete authentication flow, no flow was deleted")
+
+	var (
+		execDeleted   bool
+		execFlowFound bool
+	)
+	for _, execution := range authExecs {
+		if execution.DisplayName != nil && *execution.DisplayName == *authExecFlow.Alias {
+			execFlowFound = true
+			continue
+		}
+		if execution.ProviderID != nil && *execution.ProviderID == *authExec.Provider {
+			err = client.DeleteAuthenticationExecution(
+				context.Background(),
+				token.AccessToken,
+				cfg.GoCloak.Realm,
+				*execution.ID,
+			)
+			require.NoError(t, err, "Failed to delete authentication execution")
+			execDeleted = true
+		}
+		if execDeleted && execFlowFound {
+			break
+		}
+	}
+	require.True(t, execDeleted, "Failed to delete authentication execution, no execution was deleted")
+	require.True(t, execFlowFound, "Failed to find authentication execution flow")
```

Comment:
This should go on block up, before deleting the flow, as the flow removes the execution too. (Simply cut the lines here and move them up to line 6244). Then the test will pass.